### ISSUE TITLE
Deprecate and remove the input `contrast` prop

### ISSUE THREAD

**Issue by iansan5653** [primer/react#4355]
The `contrast` prop, when applied to `Select`, `TextInput`, or `Textarea` (one would also expect it on `TextInputWithTokens` but it's not available there) renders an input with a subtle background, somewhere between the default and disabled styling:

![screenshot of TextInput with normal, contrast, and disabled stylings](https://github.com/primer/react/assets/2294248/22612d67-cb16-4329-b2c3-369117c2e389)

In dark theme it has no effect: 

![the same screenshot in dark mode. All three inputs look the same](https://github.com/primer/react/assets/2294248/4d44fcdb-0ad1-4c46-8dc6-4c285f268327)

The documentation says this prop:

> Changes background color to a higher contrast color

But ironically it actually reduces the contrast between the text and background.

This prop is problematic because, as shown above, it can make active inputs look disabled. This can and has caused confusion for users, and so the recommendation and effort has been to avoid using it in GitHub UI (including in PVC components). We've been [considering](https://github.com/primer/react/issues/3487#issuecomment-1622579311) getting rid of it for a while, so  let's get it done.

**Comment by mperrotti** [comment]
I agree that we should get rid of the `contrast` prop. It's purpose was so you could have an input on a "secondary" background color (#f6f8fa) match the contrast of an input on a "primary" background color (#ffffff). However, it's often misused.

Here's an image for demonstration purposes:
![2 sets of inputs on primary and secondary background colors](https://github.com/primer/react/assets/2313998/9c74d61a-3d23-45d6-998d-eaeff8d28813)

**Comment by mperrotti** [comment]
I'm worried about a default transparent background breaking designs where we're only able to reach the 3:1 contrast minimum because of the white input background.

I think this use-case is pretty rare and can be safely handled with style overrides.

**Comment by lukasoppermann** [comment]
I will remove it in Figma. @mperrotti please always ping me in issues like this so I can take care of the Figma work. Thanks 🙏
